packages/atomic/src/components/result-template-components/result-template-decorators.ts

```diff
@@ -1,25 +1,35 @@
+import {Result} from '@coveo/headless';
 import {ComponentInterface, getElement} from '@stencil/core';
+import {buildCustomEvent} from '../../utils/event-utils';
+
+export class MissingResultParentError extends Error {
+  constructor(elementName: string) {
+    super(
+      `The "${elementName}" element must be the child of an "atomic-result" element.`
+    );
+  }
+}
 
 export function ResultContext() {
   return (component: ComponentInterface, resultVariable: string) => {
-    const {componentWillLoad} = component;
-    component.componentWillLoad = function () {
+    const {connectedCallback} = component;
+    component.connectedCallback = function () {
```

Comment:
Just a sanity check, but is there any chance of a race condition where we trigger the event before the parent would be ready to `Listen` to it ?

I'm assuming not, but just making sure.

Reply:
The parent is the atomic result which renders the content., from a string, previous lifecycle methods will have been launched.
It's different than the atomic-search-interface & other components relationship, which are slotted element.